src/BCFSerialize.cc

```diff
@@ -39,6 +39,11 @@ using namespace std;
 
 namespace GLnexus {
 
+static void range_check(int x, size_t y) {
+    // FIXME: this should -not- be removed in Release builds
```

Comment:
Yes, let's go ahead refactor as needed so that out-of-bounds errors anywhere will surface as Status::Invalid. Also a test case for the bounds check.